include/lcp-thumbnail.php

```diff
@@ -21,50 +21,67 @@ public function get_thumbnail($single, $thumbnail, $thumbnail_size, $force_thumb
+        array_merge(
+          get_intermediate_image_sizes(),
+          array("thumbnail", "medium", "large", "full")
+        )
+      );
+
+      if( in_array( $thumbnail_size,  $image_sizes ) ){
+        $lcp_thumb_size = $thumbnail_size;
+      } elseif( $thumbnail_size ) {
+        $lcp_thumb_size = explode(",", $thumbnail_size);
+      } else {
+        $lcp_thumb_size = 'thumbnail';
+      }
 
       if ( has_post_thumbnail($single->ID) ){
-        $image_sizes = array_unique(
-          array_merge(
-            get_intermediate_image_sizes(),
-            array("thumbnail", "medium", "large", "full")
-          )
-        );
 
-        if( in_array( $thumbnail_size,  $image_sizes ) ){
-          $lcp_thumb_size = $thumbnail_size;
-        } elseif( $thumbnail_size ) {
-          $lcp_thumb_size = explode(",", $thumbnail_size);
-        } else {
-          $lcp_thumb_size = 'thumbnail';
-        }
 
         $lcp_thumbnail = '<a href="' . esc_url(get_permalink($single->ID)) .
-                       '" title="' . esc_attr($single->post_title) . '">';
+                         '" title="' . esc_attr($single->post_title) . '">';
 
         $lcp_thumbnail .= get_the_post_thumbnail(
-            $single->ID,
-            $lcp_thumb_size,
-            array(
-                'alt' => esc_attr($single->post_title),
-                // If we have a class use it, otherwise use default
-                'class' => ( $lcp_thumb_class != null ) ? $lcp_thumb_class : 'lcp_thumbnail'
-            )
+          $single->ID,
+          $lcp_thumb_size,
+          array(
+            'alt' => esc_attr($single->post_title),
+            // If we have a class use it, otherwise use default
+            'class' => ( $lcp_thumb_class != null ) ? $lcp_thumb_class : 'lcp_thumbnail'
+          )
         );
         $lcp_thumbnail .= '</a>';
       } else {
-        // if thumbnail is requested but not found as featured image, grab first image in the content of the post
-        if ( ($force_thumbnail === 'yes'|| $force_thumbnail === 'true') && preg_match('~<img[^>]*src\s?=\s?[\'"]([^\'"]*)~i',get_the_content(), $imgMatches)) {
-          $lcp_thumbnail = '<a href="' . esc_url(get_permalink($single->ID)) .
-                         '" title="' . esc_attr($single->post_title) . '">';
+        $content = get_the_content();
+        $thumb_id = $this->get_first_content_image_id($content);;
+        if ($thumb_id) {
+          $lcp_thumbnail = '<a href="' . esc_url(get_permalink($single->ID)) . '" title="' . esc_attr($single->post_title) . '">';
+          $lcp_thumbnail .= wp_get_attachment_image(
+            $thumb_id,
+            $lcp_thumb_size,
+            array(
+              'alt' => esc_attr($single->post_title),
+              // If we have a class use it, otherwise use default
+              'class' => ( $lcp_thumb_class != null ) ? $lcp_thumb_class : 'lcp_thumbnail'
+            )
+          );
+          $lcp_thumbnail .= '</a>';
+        } else {
```

Comment:
Can you explain why you added this code while also leaving the existing code that grabs the first content image? If your intention was to refactor it and add new features why leave the old code after the `else` statement on line 70?

Moreover, note that by deleting the `if` on line 56 you make the plugin fetch an image from the content even when `force_thumbnail` is not `yes` or `true`. This is contrary to how it works now and there is no good reason to change it.

Please consider refactoring lines 56-83 so that there is one, clean procedure for forced thumbnails, including your changes of course.

I can answer any questions you might have about that, feel free to ask.

Reply:
Hello,

Thank you for your feedback, you were right, I just cleaned up.